### PR TITLE
Wallet Encrypt/Decrypt functionality exposed in the API

### PR DESCRIPTION
**Pull request by zoidbergZA**
### Motivation
This change gives consumers of the library the ability to encrypt and decrypt wallets without being locked into a specific implementation for loading and saving the encrypted data to file, local database, etc.

### Changes
- Refactored the wallet encryption logic into the module WalletEncryption.ts
- Exposed functions to encrypt a wallet and open an encrypted wallet in WalletBackend.ts
- Added a unit test for encrypting/decrypting wallets

### Note
I was unable to update the docs using 'yarn docs', the command also didn't work for me on the original branch with errors like:
`Error: ENOENT: no such file or directory, stat 'C:\github\turtlecoin-wallet-backend-js\lib\;'
    at Object.fs.statSync (fs.js:948:11)`
so the docs would still have to be updated.

### CODE REVIEW

File: lib/WalletBackend.ts [hunk 1]
```diff
@@ -1080,6 +1117,28 @@ export class WalletBackend extends EventEmitter {
...
+     * ```javascript
+     * const saved = wallet.encryptWalletToString('hunter2');
+     *
+     * ```
+     *
+     * @param password The password to encrypt the wallet with
+     *
+     * @return Returns the encrypted wallet as astring.
+     */
+    public encryptWalletToString(password: string) {
```
Comment (zpalmtree) [inline lib/WalletBackend.ts]: Can you add the return type to the typings here.

File: lib/WalletBackend.ts [hunk 2]
```diff
@@ -1080,6 +1117,28 @@ export class WalletBackend extends EventEmitter {
         return this.subWallets.getPrimaryAddress();
     }
 
+    /**
+     * encrypt the wallet using the given password. Password may be empty. Note that an empty password does not mean an
```
Comment (zpalmtree) [inline lib/WalletBackend.ts]:
```suggestion
     * Encrypt the wallet using the given password. Password may be empty. Note that an empty password does not mean an
```

File: lib/WalletBackend.ts (hunk 1 above)
Comment (zoidbergZA) [reply lib/WalletBackend.ts]: added :)

**Comment by zpalmtree** [comment]
Thanks for the PR!

**Pull request merged**